internal/pkg/service/common/etcdop/op/atomic.go

```diff
@@ -45,7 +46,22 @@ type AtomicOpInterface interface {
 }
 
 func Atomic[R any](client etcd.KV, result *R) *AtomicOp[R] {
-	return &AtomicOp[R]{client: client, result: result}
+	return &AtomicOp[R]{client: client, result: result, checkPrefixKey: true}
+}
+
+// SkipPrefixKeysCheck disables the feature.
+//
+// By default, the feature is enabled and checks that each loaded key within the Read Phase, from a prefix, exists in Write Phase.
+// This can be potentially SLOW and generate a lot of IF conditions, if there are a large number of keys in the prefix.
+// Therefore, this feature can be turned off by the method.
+//
+// Modification of a key in the prefix is always detected,
+// this feature is used to detect the deletion of a key from the prefix.
+//
+// See TestAtomicOp:GetPrefix_DeleteKey_SkipPrefixKeysCheck.
+func (v *AtomicOp[R]) SkipPrefixKeysCheck() *AtomicOp[R] {
```

Comment:
We can now check that each loaded key from the prefix exists, but the approach has some limitations.

Reply:
By default the feature is enabled, so there is by default a 100% guarantee of atomicity.